src/game/client/water/w3dwater.cpp

```diff
@@ -0,0 +1,2327 @@
+    m_riverYOffset = m_riverYOffset + 0.000165f;
+
+    if (m_riverXOffset > 1.0) {
+        m_riverXOffset = m_riverXOffset - 1.0;
+    }
+
+    if (m_riverYOffset > 1.0) {
+        m_riverYOffset = m_riverYOffset - 1.0;
+    }
+
+    if (m_riverXOffset < -1.0) {
+        m_riverXOffset = m_riverXOffset + 1.0;
+    }
+
+    if (m_riverYOffset < -1.0) {
+        m_riverYOffset = m_riverYOffset + 1.0;
+    }
+
+    m_iBumpFrame++;
+
+    if (m_iBumpFrame >= 32) {
+        m_iBumpFrame = 0;
+    }
+
+    if (lastLogicFrame != frame) {
+        if (m_doWaterGrid) {
+            if (m_meshInMotion) {
+                int x = m_gridCellsX + 1;
+                int y = m_gridCellsY + 1;
+                m_meshInMotion = false;
+                WaterMeshData *m = m_meshData;
+
+                for (int i = 0; i < y + 1; i++) {
+                    for (int j = 0; j < x + 1; j++) {
+                        if ((m->status & 1) != 0) {
+                            m->velocity = 0.93000001f * m->velocity;
+
+                            if (m->preferred_height <= m->height) {
+                                m->velocity = g_theWriteableGlobalData->m_gravity * 3.0f + m->velocity;
+                            } else {
+                                m->velocity = m->velocity - g_theWriteableGlobalData->m_gravity * 3.0f;
+                            }
+
+                            m->height = m->height + m->velocity;
+
+                            if (GameMath::Fabs(m->height - m->preferred_height) >= 1.0f
+                                || GameMath::Fabs(m->velocity) >= 1.0f) {
+                                m_meshInMotion = true;
+                            } else {
+                                m->status &= ~1;
+                                m->height = m->preferred_height;
+                                m->velocity = 0.0f;
+                            }
+                        }
+
+                        m++;
+                    }
+                }
+            }
+        }
+        lastLogicFrame = frame;
+    }
+}
+
+void WaterRenderObjClass::Replace_Skybox_Texture(Utf8String const &oldname, Utf8String const &newname)
+{
+    ((GameAssetManager *)W3DAssetManager::Get_Instance())->Replace_Prototype_Texture(m_skyBox, oldname, newname);
+
+    if (m_skyBox && m_skyBox->Class_ID() == RenderObjClass::CLASSID_MESH) {
+        MeshClass *mesh = (MeshClass *)m_skyBox;
+        MaterialInfoClass *matinfo = mesh->Get_Material_Info();
+
+        for (int i = 0; i < matinfo->Texture_Count(); i++) {
+            if (matinfo->Peek_Texture(i)) {
+                matinfo->Peek_Texture(i)->Get_Texture_Filter()->Set_U_Address_Mode(
+                    TextureFilterClass::TEXTURE_ADDRESS_CLAMP);
+                matinfo->Peek_Texture(i)->Get_Texture_Filter()->Set_V_Address_Mode(
+                    TextureFilterClass::TEXTURE_ADDRESS_CLAMP);
+            }
+        }
+
+        Ref_Ptr_Release(matinfo);

```

Comment:
where did this come from? if its a bugfix should be labeled as such

Reply:
Its there in zh wb.

Reply:
I see what I did wrong, removed it.